internal/reconciler/telemetry/reconciler.go

```diff
@@ -144,7 +144,7 @@ func (r *Reconciler) reconcileWebhook(ctx context.Context, telemetry *operatorv1
 		return nil
 	}
 
-	if !telemetry.DeletionTimestamp.IsZero() {
+	if !telemetry.DeletionTimestamp.IsZero() && !r.dependentCRsFound(ctx) {
```

Comment:
IMHO, this use case is a bit tricky. Maybe let us leave a comment here?

Reply:
Added a comment